sql/2024/fonts/README.md

```diff
@@ -1,20 +1,54 @@
 # 2024 Fonts queries
 
-<!--
-  This directory contains all of the 2024 Fonts chapter queries.
+## Resources
 
-  Each query should have a corresponding `metric_name.sql` file.
-  Note that readers are linked to this directory, so try to make the SQL file names descriptive for easy browsing.
+* 📄 [Planning document]
+* 📊 [Results sheet]
+* 📝 [Chapter content]
 
-  Analysts: if helpful, you can use this README to give additional info about the queries.
--->
+## Structure
 
-## Resources
+The queries are split by the section where they are used:
+
+* `design/` is about foundries and families,
+* `development/` is about tools and technologies, and
+* `performance/` is about hosting and serving.
+
+Each file name starts with one of the following prefixes indicating the primary
+subject of the corresponding analysis:
+
+* `fonts_` is about font files,
+* `pages_` is about HTML pages,
+* `scripts_` is about JavaScript scripts, and
+* `styles_` is about CSS style sheets.
+
+The prefix is followed by the property studied given in singular, potentially
+extended one or several suffixes narrowing down the scope, as in
+`fonts_size_by_table.sql` and `pages_link_relation.sql`.
+
+## Contents
+
+Each query starts with a preamble indicating the section, question, and
+normalization type:
+
+```sql
+-- Section: Performance
+-- Question: What is the distribution of the file size broken down by table?
+-- Normalization: Sites
+```
+
+Many queries rely on temporary functions for convenience and clarity. The
+functions appear in several queries are extracted into a common file:
+`common.sql`. Whenever any of the functions defined in `common.sql` is used by a
+query, the query has the following line at the top:
+
+```sql
+-- INCLUDE ../common.sql
+```
 
-- [📄 Planning doc][~google-doc]
-- [📊 Results sheet][~google-sheets]
-- [📝 Markdown file][~chapter-markdown]
+It signalizes that, prior to executing the query, `common.sql` has to be
+inlined.
```

Comment:
The queries are linked from the Web Almanac and this won’t be understood.

I think we should remove the common.sql files and just inline them in the files.

Reply:
Is it a hard rejection, or can this be discussed? We arguably want the queries to be easy to understand and easy to maintain. Duplicating the common functions at the top of each query will clutter the code and make it difficult to maintain, as one would have to keep track of the functions and keep them in sync.

If you want, we can replace

```sql
-- INCLUDE ../common.sql
```

with

```sql
-- INCLUDE https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/sql/2024/fonts/common.sql
```

Another alternative found be do create permanent SQL functions in a namespace in BigQuery dedicated to the chapter. Someone with the necessary permissions would have to do that.

Reply:
I think that full path of the include would be a bit better.